### ISSUE TITLE
Option pricing feature

### ISSUE THREAD

**Issue by tom-rice** [VividWeb/vivid_store#296]
**Edit: Never mind now I found this**: https://github.com/VividWeb/vivid_store/pull/279 - awesome!

I am looking to add the ability to set separate prices or +/- price modifiers for each option on a product (example screenshot/idea attached).
This way a single product could come in large or small for example, at different prices depending on size chosen.

I'm probably going to attempt to do a modification myself (and share the code with you if it's any good), unless either I missed something or I just wondered if this is something you've thought of or already planned for?

![option-pricing](https://cloud.githubusercontent.com/assets/4663011/14606147/bd904c6e-0573-11e6-93e2-d6b58b8c70c1.jpg)

**Comment by brucewyne** [comment]
Bingo. This is coming down the pipeline. Consider though that github is a "development" version though. We may make some changes before release. That said, this is likely going to be pushed to another minor version away till it's refined just a little more.

**Comment by brucewyne** [comment]
that said, test it and raise issues if it's not up to snuff.

**Comment by tom-rice** [comment]
Thanks! :)

I have already set up a new site with 3.0.1 containing various products e.t.c. and although I realise this is the development channel and needs refinement I'm hoping to try this out anyway and get it running pretty quickly as it's the one feature they're missing and really need. I can handle bugfixes as they come. Unfortunately I'm having a lot of trouble trying to upgrade vivid_store to 3.0.1.4 - errors with adding foreign key constraints and primary keys e.t.c. I don't suppose you would have any suggestions about that?

Also, I encountered a few errors with src/VividStore/Payment/Method.php:
1) It did not like that $pmID, $pmHandle and others are declared twice at the top (once as private, again as protected)
2) It was trying to use setHandle, setName, setDisplayName, setPackageID, however those don't exist, the setter methods are actually setPaymentMethodHandle, setPaymentMethodName, setPaymentMethodDisplayName, setPaymentMethodPackageID.

I was able to get around those errors by editing that file and then it did seem to run eventually with a clean install.

This was one of the errors I got stuck with on with the attempted upgrade:
`An exception occurred while executing 'ALTER TABLE VividStoreProductVariationOptionItems ADD CONSTRAINT FK_50D9C3734BB971ED FOREIGN KEY (poiID) REFERENCES VividStoreProductOptionItems (poiID) ON DELETE CASCADE': SQLSTATE[HY000]: General error: 1215 Cannot add foreign key constraint`
